.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: ci
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+jobs:
+  go:
+    strategy:
+      fail-fast: false
+      matrix:
+        container: ["cimg/go:1.14", "cimg/go:1.15"]
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ matrix.container }}
+    env:
+      TEST_RESULTS: /tmp/test-results # path to where test results will be saved
+    steps:
+      - name: Setup required permissions
+        run: sudo chmod -R 777 $GITHUB_WORKSPACE /github /__w/_temp
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: restore cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-mod
+        with:
+          path: /home/circleci/go/pkg/mod
+          key: cimg-go-pkg-mod-{{ checksum "go.sum" }}
+      - name: Precommit and Coverage Report
+        run: |
+          make ci
+          mkdir $TEST_RESULTS/
+          cp coverage.out $TEST_RESULTS/
+          cp coverage.txt $TEST_RESULTS/
+          cp coverage.html $TEST_RESULTS/
```

Comment:
Should we remove the trailing `/` here so if `TEST_RESULTS` becomes undefined it doesn't move things to `/`?

Reply:
makes sense, removed!